Makefile

```diff
@@ -32,12 +32,19 @@ test: system-check
 
 #Build the binary
 build: pre-build
-	@cd cmd/botkube;GOOS_VAL=$(shell go env GOOS) GOARCH_VAL=$(shell go env GOARCH) go build -o $(shell go env GOPATH)/bin/botkube 
+	@cd cmd/botkube;GOOS_VAL=$(shell go env GOOS) GOARCH_VAL=$(shell go env GOARCH) go build -o $(shell go env GOPATH)/bin/botkube
 	@echo "Build completed successfully"
+
 #Build the image
 container-image: pre-build
 	@echo "Building docker image"
-	@docker build --build-arg GOOS_VAL=$(shell go env GOOS) --build-arg GOARCH_VAL=$(shell go env GOARCH) -t $(IMAGE_REPO) -f build/Dockerfile --no-cache .
+	@if ! DOCKER_CLI_EXPERIMENTAL=enabled docker buildx ls | grep -q container-builder; then\
+		DOCKER_CLI_EXPERIMENTAL=enabled docker buildx create --name container-builder --use;\
+	fi
+	DOCKER_CLI_EXPERIMENTAL=enabled docker buildx build --platform "linux/amd64,linux/arm64,linux/arm/v7" \
+		-t $(IMAGE_REPO) \
+		-f build/Dockerfile \
+		. --push
```

Comment:
Is `--push` required here? Can't we just build the image locally?

Reply:
You can run --load if you'd like to build but not push the image.

Reply:
Let's do it then. Because if someone outside infracloudio org tries to build an image it will fail. Also, if someone has access to infracloudio org, then can overwrite the latest image upstream which is not desirable.

Reply:
So we can have the Makefile as --load seeing as this can manually be run from a local machine whilst keep CI to be the only --push? I can reflect those changes on my next commit.